RIOT-OS/RIOT
# cpu/stm32: Fix stm clock configuration

### Contribution description

This splits the stm32 clock tree in Kconfig making it a bit easier to look at and manage.
It exposes the clock speeds allowing enough information to match the make resolution.
It also introduces some `HAVE_` symbols to make it a bit easier to manage multiple CPU_SERIES and/ors.

There is a small fix to the make resolution for the stm32f303*, some boards were not running at the max possible clock speed.

### Testing procedure

Green murdock with `NIGHTLY=1`

Self testing based off #18771 with
```
export TEST_BOARDS=$(FEATURES_REQUIRED=cpu_stm32 make -C tests/pkg_tinyusb_cdc_msc info-boards-supported --no-print-directory)
for board in $TEST_BOARDS; do echo $board && /bin/bash -c "source .murdock; JOBS=8 compile tests/pkg_tinyusb_cdc_msc ${board}:gnu" | grep mismatch; done;
```
as well (since this should fix the USB errors)

### Issues/PRs references

Follow up #18771
